tests/console/systemd_testsuite.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's systemd tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run testsuite included in systemd sources
+# Maintainer: Thomas Blume <tblume@suse.com>
+
+use base "consoletest";
+use warnings;
+use strict;
+use testapi;
+use utils;
+
+sub run() {
+    # install systemd testsuite
+    select_console 'root-console';
+    if (check_var('VERSION', '12-SP2')) {
+        zypper_call('ar http://download.suse.de/ibs/QA:/SLE12SP2/update/ systemd-testrepo');
+    }
+
+    elsif (check_var('VERSION', '12-SP3')) {
+        zypper_call('ar http://download.suse.de/ibs/QA:/SLE12SP3/update/ systemd-testrepo');
+    }
+    else {
```

Comment:
The run of the testsuite is disabled during normal package build.
It might be useful for upstream, but calling the tests in the specfile only delays the package build and the result must be extracted from the build logs and manually reviewed.
Instead, I run the tests from freshly build source packages before a new submission via the script at:
http://beta.suse.com/private/tblume/systemd-testsuite/

But I don't quite understand, do you think it makes no sense to run these tests on openQA?
I know, there are limitations, like that currently the extended tests run within kvm or nspawn containers. This is not optimal and I'm planning to port them to run directly on the test machine.
But at the moment, this is the best we have.
Otherwise, what tests of systemd would you like to see in openQA?

Concerning SLE15 and Tumbleweed, I'm working on a submission.
There are far more extended tests in the newer systemd versions, which I first have to check and possibly to adapt to run on SUSE.